### PR TITLE
Implement function to merge UTXOs 

### LINKED ISSUE #14: Expand the transaction generator to sometimes merge UTXOs
Currently, Faucet will forever split the outputs among keys.
This is a very predictable behavior that does not reflect a live settings.

We currently only have the following variables:
- Interval: how often we send
- Count: How many transactions we generate per run

What we don't have is a way to control the structure of the transactions. We don't need to send homogenous transactions, and we definitely need to send transactions what will "merge" UTXOs (have `inputs.length > outputs.length`) as opposed to the current approach of always "splitting" (`inputs.length` < `outputs.length`).

### PR DESCRIPTION
**Pull request by kchulj**
Now that we have [agora#1345](https://github.com/bpfkorea/agora/pull/1345), a helper function can be implemented. It takes a range of `Output`s and merges them into a range of `Transaction` with a single `Input` and `Output`.

Example use-case: Whenever we have more than 200 `UTXO`s, call this function until we have less than 200.

After the function has been finalized, I will include it in the `send` process, with an appropriate bound.

Fixes #14

### CODE REVIEW

**Review by hewison-chris — APPROVED** [review]
LGTM

File: source/faucet/main.d [hunk 1]
```diff
@@ -119,6 +119,33 @@ private auto splitTx (UR) (UR utxo_rng, uint count)
...
+    Params:
+      UR = Range of tuple with an `Output` (`value`) and
+            a `Hash` (`key`), as its first and second element, respectively
+
+    Returns:
+      A transaction
+
+*******************************************************************************/
+
+private auto mergeTx (UR) (UR utxo_rng)
```
Comment (AndrejMitrovic) [inline source/faucet/main.d:139]: `auto` => `Transaction`

**Comment by kchulj** [comment]
I have tested this version with the node and it is able to create 30+ blocks.
Although the feature is very basic, it is consistent. I will build on top of this.

**Review by AndrejMitrovic — APPROVED** [review]

File: source/faucet/main.d [hunk 2]
```diff
@@ -107,7 +107,6 @@ private auto splitTx (UR) (UR utxo_rng, uint count)
     static assert (isInputRange!UR);
 
     return utxo_rng
-        .filter!(tup => tup.value.output.value >= Amount(count))
```
Comment (Geod24) [inline source/faucet/main.d:110]: Why was this removed ? This doesn't look right.

File: source/faucet/main.d [hunk 3]
```diff
@@ -118,12 +117,36 @@ private auto splitTx (UR) (UR utxo_rng, uint count)
...
+      A range of Transactions
+
+*******************************************************************************/
+
+private auto mergeTx (UR) (UR utxo_rng) @safe
+{
+    static assert (isInputRange!UR);
+
+    return utxo_rng
+            .filter!(tup => tup.value.output.value > Amount(count))
```
Comment (Geod24) [inline source/faucet/main.d:139]: And this doesn't look right either. The filter was a workaround against an issue we saw were we created too small UTXOs. Merging by definition doesn't have that issue.

Comment (kchulj) [reply source/faucet/main.d:139]: Yes, I moved the workaround  from `splitTx` to here, but this eventually needs to be replaced.
I will have to come up with a way to call merge when there exists an `Amount` that is less than `count`.

Comment (Geod24) [reply source/faucet/main.d:139]: I still don't understand: Merge doesn't have the bug, does it ? It's only `split` that has it.

Comment (kchulj) [reply source/faucet/main.d:139]: If `split` creating `Amount`s less than `count` is considered a bug, then yes.
How can we prevent `split` from creating these "remainders" when it's called?

Comment (Geod24) [reply source/faucet/main.d:139]: > If split creating Amounts less than count is considered a bug, then yes.

Yes. So why was the code removed from `split`, and put in `merge` ?

> How can we prevent `split` from creating these "remainders" when it's called?

I don't think the remainders are a problems. I actually think the filter  call was legit.

Comment (kchulj) [reply source/faucet/main.d:139]: > Yes. So why was the code removed from split, and put in merge ?

I remember, I was thinking of sending these remainders to an invalid address instead of filtering, during the `merge` process. The `filter` there should have been temporary.

But if the filter call in `split` is legit, then it works very well.
